sendAjax.js

```diff
@@ -0,0 +1,60 @@
+/*eslint strict:0*/
+/**
+ * Special test server to test sendAJAX function
+ *
+ */
+var utils = require('utils');
+
+casper.test.begin("__utils__.sendAJAX() POST Custom Headers tests", 7, {
+    setUp: function(test) {
+        this.server = require('webserver').create();
+        this.requestReceived = null;
+        this.server.listen(8585, function (request, response) {
+            requestReceived = request;
+            response.statusCode = 200;
+            response.write("");
+            response.close();
+        });
+    },
+
+    tearDown: function() {
+        this.server.close();
+        this.requestReceived = null;
+    },
+
+    test: function(test) {
+        var wsurl = 'http://127.0.0.1:8585';
+        casper.userAgent("Googlebot/2.1 (+http://www.google.com/bot.html)");
+        casper.start().then(function() {
+            this.evaluate(function(wsurl) {
+                var customData = {"requestData":"dummydata"};
+                var customSettings = {
+                    headers:{
+                        "Accept": "*/*",
+                        "Accept-Language": "pt-BR,en,*",
+                        "Content-Type": "multipart/form-data",
+                        "Host" : "192.168.0.2:8584",
+                        "Origin": "http://google.com"
+                        //For some reason underlying XMLHttpRequest.setRequestHeader() doesn't let you
+                        //set this and others Headers as well.
+                        //,"User-Agent": "Googlebot/2.1 (+http://www.google.com/bot.html)"
+                    }
```

Comment:
For security reasons, you may not set any header you like here.
See: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/setRequestHeader
https://developer.mozilla.org/en-US/docs/Glossary/Forbidden_header_name
https://developer.mozilla.org/en-US/docs/Glossary/Forbidden_response_header_name